api/src/main/java/io/minio/MinioProperties.java

```diff
@@ -31,46 +32,57 @@
 
   private static final Logger LOGGER = Logger.getLogger(MinioProperties.class.getName());
 
+  // These attributes are checked from the manifests in classpath.
+  private static final String META_INF_ATTRIB_IMPLEMENTATION_TITLE = "Implementation-Title";
+  private static final String META_INF_ATTRIB_IMPLEMENTATION_VERSION = "Implementation-Version";
+
+  // this is set from gradle
+  private static final String META_INF_ATTRIB_IMPLEMENTATION_TITLE_VALUE = "minio";
+
   private final AtomicReference<String> version = new AtomicReference<>(null);
 
   public String getVersion() {
     String result = version.get();
-    if (result == null) {
-      synchronized (INSTANCE) {
-        if (version.get() == null) {
-          try {
-            ClassLoader classLoader = getClass().getClassLoader();
-            setMinioClientJavaVersion(classLoader);
-            setDevelopmentVersion();
-          } catch (IOException e) {
-            LOGGER.log(Level.SEVERE, "IOException occured", e);
-            version.set("unknown");
-          }
-          result = version.get();
-        }
-      }
+    if (result != null) {
+      return result;
     }
-    return result;
+    setVersion();
+    return version.get();
   }
 
-  private void setDevelopmentVersion() {
-    if (version.get() == null) {
+  private synchronized void setVersion() {
+    ClassLoader classLoader = getClass().getClassLoader();
+    if (classLoader == null) {
       version.set("dev");
```

Comment:
Set `dev` as default value i.e. move above this line and remove setting it later.